contrib/platform/src/com/sun/jna/platform/win32/Shell32.java

```diff
@@ -409,5 +411,28 @@
      */
     HRESULT SetCurrentProcessExplicitAppUserModelID(WString appID);
 
+    /**
+     * Parses a Unicode command line string and returns an array of pointers to the
+     * command line arguments, along with a count of such arguments, in a way that
+     * is similar to the standard C run-time {@code argv} and {@code argc} values.
+     *
+     * @param lpCmdLine
+     *            A Unicode string that contains the full command line. If this
+     *            parameter is an empty string the function returns the path to the
+     *            current executable file.
+     * @param pNumArgs
+     *            Pointer to an {@code int} that receives the number of array
+     *            elements returned, similar to {@code argc}.
+     * @return A pointer to an array of {@link LPWSTR} values, similar to
+     *         {@code argv}. If the function fails, the return value is
+     *         {@code null}. To get extended error information, call
+     *         {@link Kernel32#GetLastError}. <br>
+     *         CommandLineToArgvW allocates a block of contiguous memory for
+     *         pointers to the argument strings, and for the argument strings
+     *         themselves; the calling application must free the memory used by the
+     *         argument list when it is no longer needed. To free the memory, use a
+     *         single call to the {@link Kernel32#LocalFree} function.
+     */
+    LPWSTR.ByReference CommandLineToArgvW(WString lpCmdLine, IntByReference pNumArgs);
```

Comment:
I suggest to return a plain pointer here - that makes it easier to handle further down and binding it here as a specialized type does not help.

```suggestion
    Pointer CommandLineToArgvW(WString lpCmdLine, IntByReference pNumArgs);
```

Reply:
Yeah, that's how [I initially did it](https://github.com/oshi/oshi/blob/84a9afc849f52c88042ff1cbcc7d7d212e1b7af0/oshi-core/src/main/java/oshi/jna/platform/windows/Shell32.java#L35).  I also incorrectly passed a `String` argument which actually worked with the function.

Reply:
I see the reason for the `WString` - it will also work with a plain `String` unless the system property `w32.ascii` is set to `true`.